code/modules/clothing/modular_armor/attachments/cape.dm

```diff
@@ -0,0 +1,90 @@
+/obj/item/armor_module/armor/cape
+	name = "6E Chameleon Cape (Long)"
+	desc = "To improve on the design of the 7E badge, this cape is capable of two colors, for all your fashion needs. It also is equiped with thermal insulators so it will double as a blanket. \n Attack with green facepaint to color. Attaches onto a uniform. Activate it to toggle the hood."
+	icon_state = "cape"
+	slot = ATTACHMENT_SLOT_CAPE
+	attachment_layer = CAPE_LAYER
+	prefered_slot = SLOT_W_UNIFORM
+	greyscale_config = /datum/greyscale_config/cape
+	flags_attach_features = ATTACH_REMOVABLE|ATTACH_SAME_ICON|ATTACH_APPLY_ON_MOB|ATTACH_ACTIVATION
+	attach_delay = 0 SECONDS
+	detach_delay = 0 SECONDS
+	secondary_color = TRUE
+	///List of slots the cape has.
+	var/list/attachments_by_slot = list(ATTACHMENT_SLOT_CAPE_HIGHLIGHT)
+	///Starting attachments that are spawned with this.
+	var/list/starting_attachments = list(/obj/item/armor_module/armor/cape_highlight)
+	///True if the hood is up, false if not.
+	var/hood = FALSE
+
+/obj/item/armor_module/armor/cape/Initialize()
+	. = ..()
+	AddComponent(/datum/component/attachment_handler, \
+	attachments_by_slot, \
+	list(
+		/obj/item/armor_module/armor/cape_highlight,
+		/obj/item/armor_module/armor/cape_highlight/half,
+		/obj/item/armor_module/armor/cape_highlight/scarf,
+	), \
+	starting_attachments = starting_attachments)
+
+/obj/item/armor_module/armor/cape/activate(mob/living/user)
+	. = ..()
+	hood = !hood
+	update_icon()
+	user.update_inv_w_uniform()
+
+/obj/item/armor_module/armor/cape/update_icon()
+	var/obj/item/armor_module/highlight = attachments_by_slot[ATTACHMENT_SLOT_CAPE_HIGHLIGHT]
+	if(hood)
+		icon_state = "cape_h"
+		highlight?.icon_state = "highlight_h"
+	else
+		icon_state = "cape"
+		highlight?.icon_state = "highlight"
+	return ..()
+
+
+/obj/item/armor_module/armor/cape/half
+	name = "6E Chameleon Cape (Half)"
+	desc = "This version of the 6E cape only has one color. However, it is five percent cheaper. It also is equiped with thermal insulators so it will double as a blanket. \n Attack with green facepaint to color. Attaches onto a uniform."
```

Comment:
```suggestion
	desc = "This version of the 6E cape only has one color. However, it is five percent cheaper. It also is equipped with thermal insulators so it will double as a blanket. \n Interact with facepaint to color. Attaches onto a uniform."
```